Assignment/Assignment/PingProcess.cs

```diff
@@ -27,40 +27,63 @@ public PingResult Run(string hostNameOrAddress)
 
     public Task<PingResult> RunTaskAsync(string hostNameOrAddress)
     {
-        throw new NotImplementedException();
+        Task<PingResult> task = Task.Run(() => Run(hostNameOrAddress)
+            );
+
+        task.Wait();
+        return task;

```

Comment:
You can just return the task here
```suggestion
        return task;
```